New warnings from compiling native maps with GCC 12.1.1

**Describe the bug**
```
g++ -g -fPIC -shared -O3 -Wall -I'/usr/lib/jvm/java-11-openjdk'/include -I'/usr/lib/jvm/java-11-openjdk'/include/linux -Ijavah  -o libaccumulo.so nativeMap/org_apache_accumulo_tserver_NativeMap.cc
In file included from nativeMap/org_apache_accumulo_tserver_NativeMap.cc:20:
In destructor ‘LocalSubKey::~LocalSubKey()’,
    inlined from ‘jlong Java_org_apache_accumulo_tserver_NativeMap_createNMI__J_3B_3B_3B_3BJZ_3I(JNIEnv*, jclass, jlong, jbyteArray, jbyteArray, jbyteArray, jbyteArray, jlong, jboolean, jintArray)’ at nativeMap/org_apache_accumulo_tserver_NativeMap.cc:97:1:
nativeMap/SubKey.h:186:19: warning: ‘void operator delete(void*, std::size_t)’ called on pointer returned from a mismatched allocation function [-Wmismatched-new-delete]
  186 |     delete(keyData);
      |                   ^
In constructor ‘SubKey::SubKey(LinkedBlockAllocator*, JNIEnv*, jbyteArray, jbyteArray, jbyteArray, jlong, jboolean, int32_t)’,
    inlined from ‘LocalSubKey::LocalSubKey(JNIEnv*, jbyteArray, jbyteArray, jbyteArray, jlong, jboolean)’ at nativeMap/SubKey.h:183:143,
    inlined from ‘jlong Java_org_apache_accumulo_tserver_NativeMap_createNMI__J_3B_3B_3B_3BJZ_3I(JNIEnv*, jclass, jlong, jbyteArray, jbyteArray, jbyteArray, jbyteArray, jlong, jboolean, jintArray)’ at nativeMap/org_apache_accumulo_tserver_NativeMap.cc:85:42:
nativeMap/SubKey.h:88:39: note: returned from ‘void* operator new [](std::size_t)’
   88 |         keyData = new uint8_t[totalLen];
      |                                       ^
In file included from nativeMap/SubKey.h:21:
In destructor ‘LocalField::~LocalField()’,
    inlined from ‘jlong Java_org_apache_accumulo_tserver_NativeMap_createNMI__J_3B_3B_3B_3BJZ_3I(JNIEnv*, jclass, jlong, jbyteArray, jbyteArray, jbyteArray, jbyteArray, jlong, jboolean, jintArray)’ at nativeMap/org_apache_accumulo_tserver_NativeMap.cc:97:1:
nativeMap/Field.h:141:17: warning: ‘void operator delete(void*, std::size_t)’ called on pointer returned from a mismatched allocation function [-Wmismatched-new-delete]
  141 |     delete(field);
      |                 ^
In constructor ‘LocalField::LocalField(JNIEnv*, jbyteArray)’,
    inlined from ‘jlong Java_org_apache_accumulo_tserver_NativeMap_createNMI__J_3B_3B_3B_3BJZ_3I(JNIEnv*, jclass, jlong, jbyteArray, jbyteArray, jbyteArray, jbyteArray, jlong, jboolean, jintArray)’ at nativeMap/org_apache_accumulo_tserver_NativeMap.cc:84:24:
nativeMap/Field.h:136:27: note: returned from ‘void* operator new [](std::size_t)’
  136 |     field= new uint8_t[len];
      |                           ^
In destructor ‘LocalSubKey::~LocalSubKey()’,
    inlined from ‘jlong Java_org_apache_accumulo_tserver_NativeMap_createNMI__J_3B_3B_3B_3BJZ_3I(JNIEnv*, jclass, jlong, jbyteArray, jbyteArray, jbyteArray, jbyteArray, jlong, jboolean, jintArray)’ at nativeMap/org_apache_accumulo_tserver_NativeMap.cc:97:1:
nativeMap/SubKey.h:186:19: warning: ‘void operator delete(void*, std::size_t)’ called on pointer returned from a mismatched allocation function [-Wmismatched-new-delete]
  186 |     delete(keyData);
      |                   ^
In constructor ‘SubKey::SubKey(LinkedBlockAllocator*, JNIEnv*, jbyteArray, jbyteArray, jbyteArray, jlong, jboolean, int32_t)’,
    inlined from ‘LocalSubKey::LocalSubKey(JNIEnv*, jbyteArray, jbyteArray, jbyteArray, jlong, jboolean)’ at nativeMap/SubKey.h:183:143,
    inlined from ‘jlong Java_org_apache_accumulo_tserver_NativeMap_createNMI__J_3B_3B_3B_3BJZ_3I(JNIEnv*, jclass, jlong, jbyteArray, jbyteArray, jbyteArray, jbyteArray, jlong, jboolean, jintArray)’ at nativeMap/org_apache_accumulo_tserver_NativeMap.cc:85:42:
nativeMap/SubKey.h:88:39: note: returned from ‘void* operator new [](std::size_t)’
   88 |         keyData = new uint8_t[totalLen];
      |                                       ^
In destructor ‘LocalField::~LocalField()’,
    inlined from ‘jlong Java_org_apache_accumulo_tserver_NativeMap_createNMI__J_3B_3B_3B_3BJZ_3I(JNIEnv*, jclass, jlong, jbyteArray, jbyteArray, jbyteArray, jbyteArray, jlong, jboolean, jintArray)’ at nativeMap/org_apache_accumulo_tserver_NativeMap.cc:97:1:
nativeMap/Field.h:141:17: warning: ‘void operator delete(void*, std::size_t)’ called on pointer returned from a mismatched allocation function [-Wmismatched-new-delete]
  141 |     delete(field);
      |                 ^
In constructor ‘LocalField::LocalField(JNIEnv*, jbyteArray)’,
    inlined from ‘jlong Java_org_apache_accumulo_tserver_NativeMap_createNMI__J_3B_3B_3B_3BJZ_3I(JNIEnv*, jclass, jlong, jbyteArray, jbyteArray, jbyteArray, jbyteArray, jlong, jboolean, jintArray)’ at nativeMap/org_apache_accumulo_tserver_NativeMap.cc:84:24:
nativeMap/Field.h:136:27: note: returned from ‘void* operator new [](std::size_t)’
  136 |     field= new uint8_t[len];
      |                           ^
```

**Versions (OS, Maven, Java, and others, as appropriate):**
 - Affected version(s) of this project: 2.1.0-SNAPSHOT
 - OS: Fedora 36, GNU Linux  5.17.8-300.fc36.x86_64
 - Others: gcc-c++-12.1.1-1.fc36.x86_64

**To Reproduce**
Steps to reproduce the behavior (or a link to an example repository that reproduces the problem):
1. `mvn clean package -DskipTests`

**Expected behavior**
Native maps should compile without warnings

**Additional context**
I only noticed this after I upgrade to Fedora 36, with GCC 12. Fedora 35 had GCC 11 and I didn't notice it there.

--- a/server/native/src/main/c++/nativeMap/Field.h
+++ b/server/native/src/main/c++/nativeMap/Field.h
@@ -138,7 +138,7 @@ struct LocalField : public Field {
   }
 
   ~LocalField(){
-    delete(field);
+    delete[] field;
   }
 };
 #endif

--- a/server/native/src/main/c++/nativeMap/SubKey.h
+++ b/server/native/src/main/c++/nativeMap/SubKey.h
@@ -183,7 +183,7 @@ struct LocalSubKey : public SubKey {
   LocalSubKey(JNIEnv *env, jbyteArray cf, jbyteArray cq, jbyteArray cv, jlong ts, jboolean del):SubKey(NULL, env, cf, cq, cv, ts, del, INT_MAX){}
 
   ~LocalSubKey(){
-    delete(keyData);
+    delete[] keyData;
   }
 };
 

